hdf5/src/hl/chunks.rs

```diff
@@ -0,0 +1,181 @@
+use crate::internal_prelude::*;
```

Comment:
Wondering, is there a good reason for moving this into a `chunks` module? (no hard preference though...)

Reply:
`dataset.rs` is 1000 lines already so keeping it short in a separate file. I'd be happy to move it back, makes no difference to me

Reply:
Agreed, let's leave it be for now. (ideally, things should be split up in many, many more modules...)